core/objlist.hpp

```diff
@@ -208,6 +208,19 @@ class ObjList : public ObjListBase {
         return (*attrlist);
     }
 
+    template<typename AttrT>
+    AttrList<ObjT, AttrT>& create_attrlist(const std::string& attr_name, AttrT default_attr) {
+        if (attrlist_map.find(attr_name) != attrlist_map.end()) {
+            throw base::HuskyException("ObjList<T>::create_attrlist error: name already exists");
+        }
+        auto* attrlist = new AttrList<ObjT, AttrT>(&objlist_data_);
```

Comment:
Looks like that you still need to use the default constructor. I think we should consider if the AttrT doesn't have default constructor.

Maybe we should provide constructor for AttrList like `AttrList<ObjT, AttrT>(&objlist_data_, default_attr)`.

You can also refer to the constructor of `std::vector`.